airflow/providers/google/cloud/operators/dataplex.py

```diff
@@ -610,3 +612,740 @@ def execute(self, context: Context) -> None:
+        retry: Retry | _MethodDefault = DEFAULT,
+        timeout: float | None = None,
+        metadata: Sequence[tuple[str, str]] = (),
+        gcp_conn_id: str = "google_cloud_default",
+        impersonation_chain: str | Sequence[str] | None = None,
+        fail_on_dq_failure: bool = False,
+        wait_for_results: bool = True,
+        fail_on_job_failure: bool = False,
+        run_timeout: float | None = None,
+        *args,
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.project_id = project_id
+        self.region = region
+        self.data_scan_id = data_scan_id
+        self.job_id = job_id
+        self.api_version = api_version
+        self.retry = retry
+        self.timeout = timeout
+        self.metadata = metadata
+        self.gcp_conn_id = gcp_conn_id
+        self.impersonation_chain = impersonation_chain
+        self.fail_on_dq_failure = fail_on_dq_failure
+        self.fail_on_job_failure = fail_on_job_failure
+        self.wait_for_results = wait_for_results
+        self.run_timeout = run_timeout
+
+    def execute(self, context: Context) -> dict:
+        hook = DataplexHook(
+            gcp_conn_id=self.gcp_conn_id,
+            api_version=self.api_version,
+            impersonation_chain=self.impersonation_chain,
+        )
+        # fetch the last job
+        if not self.job_id:
+            jobs = hook.list_data_scan_jobs(
+                project_id=self.project_id,
+                region=self.region,
+                data_scan_id=self.data_scan_id,
+                retry=self.retry,
+                timeout=self.timeout,
+                metadata=self.metadata,
+            )
+            job_ids = [DataScanJob.to_dict(job) for job in jobs]
+            if not job_ids:
+                raise AirflowException("There are no jobs, you should create one before.")
+            job_id = job_ids[0]["name"]
+            self.job_id = job_id.split("/")[-1]
+
+        if self.wait_for_results:
+            hook.wait_for_data_scan_job(
+                job_id=self.job_id,
+                data_scan_id=self.data_scan_id,
+                project_id=self.project_id,
+                region=self.region,
+                fail_on_dq_failure=self.fail_on_dq_failure,
+                fail_on_job_failure=self.fail_on_job_failure,
+                fail_on_timeout=False,
+                run_timeout=self.run_timeout,
+            )
+
+        job = hook.get_data_scan_job(
+            project_id=self.project_id,
+            region=self.region,
+            job_id=self.job_id,
+            data_scan_id=self.data_scan_id,
+            retry=self.retry,
+            timeout=self.timeout,
+            metadata=self.metadata,
+        )
+
+        if self.fail_on_dq_failure:
+            if job.state == DataScanJob.State.SUCCEEDED and not job.data_quality_result.passed:
+                raise AirflowException(f"Data Quality failed: {self.data_scan_id}")
```

Comment:
What will be the output if the job has succeeded and job.data_quality_result.passed?

Reply:
returns job that contains:
{
    ...
    "state": "SUCCEEDED",
    "type": "DATA_QUALITY",
    "dataQualityResult": {
        ...,
        "rowCount": "2",
        "passed": true
    }
}